src/scheduler/mod.rs

```diff
@@ -920,9 +934,19 @@ pub(crate) fn add_current_core() {
 			core_id.try_into().unwrap(),
 			&CoreLocal::get().scheduler_input,
 		);
+		#[cfg(target_arch = "x86_64")]
+		CORE_HLT_STATE
+			.lock()
+			.insert(core_id.try_into().unwrap(), &CoreLocal::get().hlt);
 	}
 }
 
+#[inline]
+#[cfg(all(target_arch = "x86_64", feature = "smp"))]
+pub(crate) fn get_core_hlt_state(core_id: CoreId) -> bool {
```

Comment:
```suggestion
pub(crate) fn take_core_hlt_state(core_id: CoreId) -> bool {
```

Since this is where `hlt` is set to `false` again, which is stronger than just getting.

Reply:
I applied the change.